### PR TITLE
Feature/ Adding event in calendar.

### PR DESCRIPTION
**Pull request by Mekhak11**

### CODE REVIEW

**Comment by Mekhak11** [comment]
Uploading addEvent.mp4…

**Comment by Mekhak11** [comment]
https://vioo.cc/v/3LPCW

File: Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift [hunk 1]
```diff
@@ -108,6 +117,9 @@ extension BirthDayEditCommonView {
   private var doneButton: some View {
     Button {
       doneAction(birthdayData)
+      if isAddingEvent {
+        CalendarManager.shared.addEventAction(event: birthdayData)
```
Comment (nellyAslanyan) [inline Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift]: are you accessing permission to add calendar?

Comment (Mekhak11) [reply Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift]: Yes

**Comment by nellyAslanyan** [comment]
Please fix conflicts and edit or remove uploading comment

File: Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift [hunk 2]
```diff
@@ -14,13 +17,19 @@ struct BirthDayEditCommonView: View {
...
   
+  var isCreating: Bool
   var doneAction: (BirthdayModel) -> ()
   var columns = [GridItem(.flexible()),GridItem(.flexible()),GridItem(.flexible())]
   
   var body: some View {
     content
       .background(Color.lightPink)
+      .sheet(isPresented: $openCalendar, content: {
+        EventEditViewController(birthday: $birthdayData, eventStore: EKEventStore())
```
Comment (Aslanyan99n) [inline Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift]: Please change to multiline

File: Birthday/Birthday/Utils/Managers/CalendarManager.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  }
+  
+  func updateUIViewController(_ uiViewController: EKEventEditViewController, context: Context) { }
+  
+  func makeCoordinator() -> Coordinator {
+    return Coordinator(self)
+  }
+  
+  class Coordinator: NSObject, EKEventEditViewDelegate {
+    var parent: EventEditViewController
```
Comment (Aslanyan99n) [inline Birthday/Birthday/Utils/Managers/CalendarManager.swift:41]: add spaces between { }

File: Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift [hunk 4]
```diff
@@ -6,6 +6,9 @@
 //
 
 import SwiftUI
+import EventKit
+import EventKitUI
+
```
Comment (nellyAslanyan) [inline Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift:11]: extra line

File: Birthday/Birthday/Utils/Managers/CalendarManager.swift [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+//  CalendarManager.swift
+//  Birthday
+//
+//  Created by MEKHAK GHAPANTSYAN on 28.10.24.
+//
+
+import Foundation
+import EventKit
+import SwiftUI
+import EventKitUI
```
Comment (nellyAslanyan) [inline Birthday/Birthday/Utils/Managers/CalendarManager.swift:9]: lets keep alphabetical order

File: Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift (hunk 2 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift]: fixed.

File: Birthday/Birthday/Utils/Managers/CalendarManager.swift (hunk 3 above)
Comment (Mekhak11) [reply Birthday/Birthday/Utils/Managers/CalendarManager.swift:41]: fixed.

File: Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift (hunk 4 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Home/Birthday Details/View/BirthDayEditCommonView.swift:11]: fixed

File: Birthday/Birthday/Utils/Managers/CalendarManager.swift (hunk 5 above)
Comment (Mekhak11) [reply Birthday/Birthday/Utils/Managers/CalendarManager.swift:9]: fixed

**Review by Aslanyan99n — DISMISSED** [review]

**Review by nellyAslanyan — APPROVED** [review]

**Pull request merged**